docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,204 @@
+
+### JSON Schema Definition
+
+Based on the above parameters, this may be an example of a JSON Schema for WordPress:
+
+```json
+{ 
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+
+  "properties": {
+    "wordpressUsername": { "type": "string", "title": "Username", "form": "username" },
+    "wordpressPassword": { "type": "string", "title": "Password", "form": "password" },
+    "wordpressEmail": { "type": "string", "title": "Email", "form": "email" },
+    "mariadb": { 
+      "type": "object",
+      "properties": {
+        "enabled": { "type": "boolean", "title": "Enable External Database" }
+      }
+    },
+    "externalDatabase": {
+      "type": "object",
+      "title": "External Database Details",
+      "form": "externalDatabase",
+      "properties": {
+        "host": { "type": "string" },
+        "user": { "type": "string" },
+        "password": { "type": "string" },
+        "database": { "type": "string" },
+        "port": { "type": "integer" }
+      }
+    },
+    "persistence": {
+      "type": "object",
+      "properties": {
+        "size": {
+          "type": "string",
+          "title": "Disk Size",
+          "form": "diskSize",
+          "anyOf": [
+            {"pattern": ".*"},
+            {"enum": ["10Gi", "50Gi", "100Gi"]},
+          ]
+        }
+      }
+    },
+    "resources": {
+      "type": "object",
+      "title": "Required Resources",
+      "description": "Deployment resource requests",
+      "form": "resources",
+      "properties": {
+        "requests": {
+          "type": "object",
+          "properties": {
+            "memory": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["128Mi", "512Mi", "1Gi"] }
+              ]
+            },
+            "cpu": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["100m", "300m", "500m"] }
+              ]
+            }
+          }
+        },
+      }
+    },
+    "enableMetrics": { "type": "boolean", "title": "Enable Metrics", "form": "enableMetrics" },
+    "replicas": { "type": "integer", "title": "Number of Replicas", "form": "replicas" }
+  },
+
+  "required": ["wordpressUsername", "wordpressEmail", "persistence", "resources"],
+}
+```
+
+This `values.schema.json` is a valid JSON schema for the `values.yaml` of WordPress. It can be more complete since it's possible to include every single parameter of the values.yaml file.
+
+Note that in order to be able to provide a richer presentation of the form we are including the tags `form={}` with the set of well-known parameters. This allows us to know which parameters we need to represent in the form and how to map them to valid values. In case there is a parameter that is not known we should be able to parse it as a generic input.
```

Comment:
Great. So any field with a `form` key is to be included on the form, using input matching the schema (ie. those with `enum` will be selects, those with `string` will be input, etc). The well-known ones (like `diskSize`) can be further customised or have useful graphic labels or whatever, but the generic rendering should be a good start.

Not sure if it's also worth supporting something like `tabIndex=1` or similar to enable chart authors to define the form field layout order?

Reply:
> Not sure if it's also worth supporting something like `tabIndex=1` or similar to enable chart authors to define the form field layout order?

That could be an interesting feature but probably it's better to keep it as simple as possible for the moment for the maintainers. If they require that property we can add it afterwards.